packages/flutter/lib/src/material/tooltip.dart

```diff
@@ -169,7 +169,12 @@ class Tooltip extends StatefulWidget {
     super.key,
     this.message,
     this.richMessage,
+    @Deprecated(
+      'Use Tooltip.constraints instead. '
+      'This feature was deprecated after v3.30.0-0.0.pre.',
+    )
     this.height,
```

Comment:
we should reimplement this with constraints
in initializer, if constraints is null but height is not.

this.constaints = BoxConstraints.tightFor(height: height);

we should also assert only one of them is provided

Reply:
> we should reimplement this with constraints
> in initializer, if constraints is null but height is not.
>
> this.constaints = BoxConstraints.tightFor(height: height);

I'd prefer to do it this way too, but it would require deconstifying the tooltip's constructor, so I've decided to handle this during widget build.

> we should also assert only one of them is provided

Done for both the tooltip and theme data!